logger.c

```diff
@@ -330,6 +330,70 @@ static int _logger_parse_prx_raw(logentry *e, char *scratch) {
             le->elapsed, le->cmd, le->type, le->code);
     return total;
 }
+
+// TODO (v2): the length caps here are all magic numbers. Haven't thought of
+// something yet that I like better.
+// Should at least make a define to the max log len (1024) and do some math
+// here.
+static void _logger_log_proxy_req(logentry *e, const entry_details *d, const void *entry, va_list ap) {
+    char *req = va_arg(ap, char *);
+    int reqlen = va_arg(ap, uint32_t);
+    long elapsed = va_arg(ap, long);
+    unsigned short type = va_arg(ap, int);
+    unsigned short code = va_arg(ap, int);
+    int status = va_arg(ap, int);
+    char *detail = va_arg(ap, char *);
+    size_t dlen = va_arg(ap, size_t);
+    char *be_name = va_arg(ap, char *);
+    char *be_port = va_arg(ap, char *);
+
+    struct logentry_proxy_req *le = (void *)e->data;
+    le->type = type;
+    le->code = code;
+    le->status = status;
+    le->dlen = dlen;
+    le->elapsed = elapsed;
+    le->be_namelen = strlen(be_name);
+    le->be_portlen = strlen(be_port);
+    char *data = le->data;
+    if (req[reqlen-2] == '\r') {
+        reqlen -= 2;
+    } else {
+        reqlen--;
+    }
+    if (reqlen > 300) {
+        reqlen = 300;
+    }
+    if (dlen > 150) {
+        dlen = 150;
+    }
+    // be_namelen and be_portlen can't be longer than 255+6
+    le->reqlen = reqlen;
+    memcpy(data, req, reqlen);
+    data += reqlen;
+    memcpy(data, detail, dlen);
+    data += dlen;
+    memcpy(data, be_name, le->be_namelen);
+    data += le->be_namelen;
+    memcpy(data, be_port, le->be_portlen);
+    e->size = sizeof(struct logentry_proxy_req) + reqlen + dlen;
+}
+
+static int _logger_parse_prx_req(logentry *e, char *scratch) {
+    int total;
+    struct logentry_proxy_req *le = (void *)e->data;
+
+    // TODO: encode detail / req
+    total = snprintf(scratch, LOGGER_PARSE_SCRATCH,
+            "ts=%d.%d gid=%llu type=proxy_req elapsed=%lu type=%d code=%d status=%d be=%.*s:%.*s detail=%.*s req=%.*s\n",
```

Comment:
Thanks for putting this patch out @dormando !

Sharing some information I found while testing this out, I notice a lot of info getting truncated:
1. The BE ips and ports are getting truncated for many of the calls. 
2. The `req` doesn't seem to print the entire command. It sometimes prints some of the information, but it doesn't contain the prefix and key.

Eg logs:
`ts=1649199095.50545 gid=489377 type=proxy_req elapsed=132 type=100 code=15 status=0 be=10.1:w detail=failover_hit req=get       2321742

ts=1649199095.50701 gid=489378 type=proxy_req elapsed=136 type=100 code=15 status=0 be=10.1:w detail=failover_hit req=get       2321824

ts=1649199095.51151 gid=489379 type=proxy_req elapsed=125 type=100 code=15 status=0 be=10.1:w detail=failover_hit req=get       2321743

ts=1649199095.51330 gid=489380 type=proxy_req elapsed=157 type=100 code=15 status=0 be=<valid-ip>:11211 detail=failover_hit req=get       2321825`

Note, the last logline did have a valid IP, but I just replaced it with <valid-ip>.

Note also that I also am not using startfile.lua. We've been using proxylibs as a starting point for our Lua files and I ported the use of log_req there.

Reply:
I'm going to play around with the Lua to not print on every sampled GET but one of the things I suspect is: Due to this logline being printed so frequently, maybe we're overwriting our circular logger buffer.

Reply:
log issue should be fixed. forgot to update the entry size when I added the backend ip/port.

Reply:
Oh wait, my bad, the key does get printed sometimes. It sometimes gets truncated.
2321742, 2321824, etc. in the logs above are the key names.

Reply:
keep an eye out for an off-by-one-or-two bug still. insomnia has left my ability to math out to lunch.